plugin/loinc-transformation-maven-plugin/src/main/java/dev/ikm/maven/LoincTransformationMojo.java

```diff
@@ -355,179 +490,147 @@ private void createLoincRowConcept(Composer composer, String[] columns) {
-                                .with(relatedNames2)
-                                .with(displayName)
-                                .with(definitionDescription)
-                        );
-            });
+            session.compose((SemanticAssembler semanticAssembler) -> semanticAssembler
+                    .semantic(semantic)
+                    .pattern(TinkarTerm.DESCRIPTION_PATTERN)
+                    .reference(concept)
+                    .fieldValues(fieldValues -> fieldValues
+                            .with(TinkarTerm.ENGLISH_LANGUAGE)
+                            .with(description)
+                            .with(TinkarTerm.DESCRIPTION_NOT_CASE_SENSITIVE)
+                            .with(descriptionType)
+                    ));
         } catch (Exception e) {
-            LOG.error("Error creating description semantic for concept: " + concept, e);
+            LOG.error("Error creating " + typeStr + " description semantic for concept: " + concept, e);
         }
     }
 
     /**
      * Creates an identifier semantic based on LOINC_NUM.
      */
-    private void createIdentifierSemantic(Session session, EntityProxy.Concept concept, String loincNum) {
+    private void createIdentifierSemantic(Session session, EntityProxy.Concept concept, String identifier) {
+        EntityProxy.Concept identifierSource = LoincUtility.getLoincNumConcept(namespace);
         try {
             session.compose((SemanticAssembler assembler) -> {
                 assembler.semantic(EntityProxy.Semantic.make(
-                                PublicIds.of(UuidT5Generator.get(namespace, concept.toString() + loincNum))))
+                                PublicIds.of(UuidT5Generator.get(namespace, concept.toString() + identifier))))
                         .pattern(IDENTIFIER_PATTERN)
                         .reference(concept)
-                        .fieldValues(fv -> fv.with(loincNum));
+                        .fieldValues(fv -> fv
+                                .with(identifierSource)
+                                .with(identifier));
             });
         } catch (Exception e) {
             LOG.error("Error creating identifier semantic for concept: " + concept, e);
         }
     }
 
+    private void createAxiomSemanticForPartConcept(Session session, EntityProxy.Concept concept, String partTypeName) {
+        EntityProxy.Semantic axiomSemantic = EntityProxy.Semantic.make(PublicIds.of(UuidT5Generator.get(namespace, concept.toString() + partTypeName)));
+        EntityProxy.Concept parentConcept = LoincUtility.getParentForPartType(namespace, partTypeName);
+        try {
+            if (parentConcept!= null) {
+                session.compose(new StatedAxiom()
+                                .semantic(axiomSemantic)
+                                .isA(parentConcept),
+                        concept);
+            } else {
+                if (parentConcept!= null) {
```

Comment:
I don't think we can ever get into this second if (parentConcept != null) block.